src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java

```diff
@@ -688,7 +688,13 @@ public void scan(JCTree jcTree) {
                             jcIdent.sym.owner instanceof Symbol.MethodSymbol &&
                             ((Symbol.MethodSymbol) jcIdent.sym.owner).params.contains(jcIdent.sym) &&
                             seenParams.add(jcIdent.sym)) {
-                            afterParams.add(beforeParamOrder.get(((Symbol.MethodSymbol) jcIdent.sym.owner).params.indexOf(jcIdent.sym)));
+                            Integer idx = beforeParamOrder.get(((Symbol.MethodSymbol) jcIdent.sym.owner).params.indexOf(jcIdent.sym));
+                            // TODO fix ugly hack
+                            if (idx == null && beforeParamOrder.size() == 1) {
+                                afterParams.add(0);
+                            } else {
+                                afterParams.add(idx);
+                            }
```

Comment:
This is likely a result of having before/after templates with different types, and/or different number of arguments, meaning the look up fails. Possibly we'll want to look at the names of the variables used, but that's something to explore.

Reply:
Reverted it, I understood now what I did wrong. There is no support for beforeTemplates where you do not use the incoming argument. For example:

```java
@BeforeTemplate
String string(String hello) {
    return "world"'
}
```

But it does not make any sense at all to support such a `@BeforeTemplate`. So I don't think we should ponder about this any longer.